src/main/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfig.java

```diff
@@ -0,0 +1,300 @@
+      || containsMacro(ServiceNowConstants.PROPERTY_OPERATION) || containsMacro(PROPERTY_EXTERNAL_ID_FIELD)) {
+      return;
+    }
+
+    Schema tableSchema = getTableMetaData(tableName, this).getSchema();
+    Set<String> tableFields = getTableFields(tableSchema);
+
+    Set<String> inputFields = schema.getFields()
+      .stream()
+      .map(Schema.Field::getName)
+      .collect(Collectors.toSet());
+
+    String externalIdFieldName = null;
+    switch (operation) {
+      case ServiceNowConstants.INSERT_OPERATION:
+        break;
+      case ServiceNowConstants.UPDATE_OPERATION:
+        externalIdFieldName = SERVICENOW_ID_FIELD;
+        break;
+      default:
+        collector.addFailure("Unsupported value for operation: " + operation, null)
+          .withConfigProperty(ServiceNowConstants.PROPERTY_OPERATION);
+    }
+
+    if (externalIdFieldName != null && !inputFields.remove(externalIdFieldName)) {
+      collector.addFailure(String.format("Schema must contain external id field '%s'", externalIdFieldName), null)
+        .withConfigProperty(ServiceNowSinkConfig.PROPERTY_EXTERNAL_ID_FIELD);
+    }
+    inputFields.removeAll(tableFields);
+
+    if (!inputFields.isEmpty()) {
+      for (String inputField : inputFields) {
+        collector.addFailure(
+            String.format("Field '%s' is not present or not creatable in target ServiceNow table.", inputField), null)
+          .withInputSchemaField(inputField);
+      }
+    }
+    validateInputSchema(schema, collector);
+  }
+
+  @Nullable
+  private ServiceNowTableInfo getTableMetaData(String tableName, ServiceNowSinkConfig conf) {
+    // Call API to fetch first record from the table
+    ServiceNowTableAPIClientImpl restApi = new ServiceNowTableAPIClientImpl(conf);
+    ServiceNowTableDataResponse response = restApi.fetchTableSchema(tableName, SourceValueType.SHOW_DISPLAY_VALUE,
+      null, null, true);
+    if (response == null) {
+      return null;
+    }
+    List<ServiceNowColumn> columns = response.getColumns();
+    if (columns == null || columns.isEmpty()) {
+      return null;
+    }
+    Schema schema = SchemaBuilder.constructSchema(tableName, columns);
+    return new ServiceNowTableInfo(tableName, schema, response.getTotalRecordCount());
+  }
+
+  private Set<String> getTableFields(Schema schema) {
+    Set<String> tableFields = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+    for (Schema.Field field : schema.getFields()) {
+      tableFields.add(field.getName());
+    }
+    return tableFields;
+  }
+
+  /**
+   * Checks that input schema is correct. Which means:
+   * 1. All fields in it are present in table
+   * 2. Field types are in accordance with the actual types in table.
+   *
+   * @param schema input schema to check
+   */
+  private void validateInputSchema(Schema schema, FailureCollector collector) {
+    Schema tableActualSchema = getTableMetaData(tableName, this).getSchema();
+    checkCompatibility(tableActualSchema, schema, collector, false);
```

Comment:
why is checkNullable param set to false here? Why would we want to check nullability in some cases but not others?

Reply:
The `null` check is not required as ServiceNow doesn't provide any metadata API to tell whether a field is nullable or not.
so removing the checkNullable param 23fe7afc2aaefe65b4e0108e8a5e01afa5be5cee

Reply:
Can you update the javadoc for this method (remove references to nullability checks)? https://github.com/data-integrations/servicenow-plugins/pull/31/files#diff-cbfc90dc038c25ca1a7580ba637464907f3fa6828e194d5359726a282ab39104R258-R259

Reply:
Updated Javadoc. cc191dbb6648be9ee473609d2116f6ad050c8d9c